components/raftstore/src/store/snap.rs

```diff
@@ -1136,6 +1142,14 @@ impl Snapshot {
         self.cf_files.iter().map(|cf| cf.kv_count).sum()
     }
 
+    pub fn set_hold_tmp_files(&mut self, v: bool) {
+        self.hold_tmp_files = v;
+    }
+
+    pub fn hold_tmp_files(&self) -> bool {
```

Comment:
What's this?

Reply:
When calling `get_snapshot_for_building`, the returned `Snapshot` will have `hold_tmp_files  = true`, which will lead to the snapshot file being deleted when drop `Snapshot`. If the file is deleted, then the snapshot can't pass validation when we received the empty snapshot that we send to ourselves.
By setting this `hold_tmp_files  = false` here, we can prevent deleting the snapshot file. Though it may result in some `gen_xxxx_yy_zz.meta.tmp` files which has 0-bytes, but I think I can delete the file later.

Reply:
Why the file will be deleted? Isn't it sent?

Reply:
@BusyJay Do you mean we should use the following sequence:
1. Call `get_snapshot_for_building` for a `Snapshot` object
2. Create meta file and cf files according to the `Snapshot` object
3. Build `SnapshotMetadata` and `RaftSnapshotData` and move it into `eraftpb::Snapshot`
4. Send the `eraftpb::Snapshot` with Transport
5. Delete the `Snapshot` object.

Reply:
The problem is that in `Snapshot::delete`, it will delete `self.meta_file.path`.
```Rust
delete_file_if_exist(&self.meta_file.path).unwrap();
```
And when receiving snapshot, it will call `snapshot.exists()` which will check if `self.meta_file.path` is present. However, it is deleted by `Snapshot::delete`
```Rust
pub fn exists(&self) -> bool {
    self.cf_files.iter().all(|cf_file| {
        cf_file.size.is_empty()
            || (cf_file
                .file_paths()
                .iter()
                .all(|file_path| file_exists(Path::new(file_path))))
    }) && file_exists(&self.meta_file.path)
}
```

Reply:
The meta path of generating and receiving snapshots are different. Why that matters?